llvm/test/SYCLLowerIR/PropagateAspectUsagePass/composite_types4.ll

```diff
@@ -0,0 +1,80 @@
+; RUN: opt --PropagateAspectUsage < %s -S | FileCheck %s
+;
+; Test checks complex types graphs.
+
+;     ----> A
+;     |     |
+;     |     v
+;    *|     B
+;     |    / \
+;     |   v   v
+;     E<--C   D
+;     |   ^
+;    *|   |*
+;     v   |
+;     F-->G
+;       *
+;
+
+%A = type { %B }
+%B = type { %C, %D }
+%C = type { %E }
+%D = type { i32 }
+%E = type { %A*, %F* }
+%F = type { %G* }
+%G = type { %C* }
+
+
+; CHECK-DAG: dso_local spir_kernel void @kernel1() !intel_used_aspects !2 {
+define dso_local spir_kernel void @kernel1() {
+  %tmp = alloca %A
+  ret void
+}
+
+; CHECK-DAG: dso_local spir_kernel void @kernel2() !intel_used_aspects !2 {
+define dso_local spir_kernel void @kernel2() {
+  %tmp = alloca %C
+  ret void
+}
+
+; CHECK-DAG: dso_local spir_kernel void @kernel3() !intel_used_aspects !2 {
+define dso_local spir_kernel void @kernel3() {
+  %tmp = alloca %F
+  ret void
+}
+
+;     -----> A1
+;     |      |
+;     |      v
+;    *|      B1
+;     |     / \
+;     |    v   v
+;     E1<--C1   D
+;     |    ^
+;    *|    |*
+;     v    |
+;     F1-->G1-->D1
+;       *
```

Comment:
I suggest that we remove `D` from this test case and check aspects of type `A` to test aspect propagation through a more interesting path